src/formatDate.js

```diff
@@ -8,7 +8,29 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const object = {};
+  const splitDate = date.split(fromFormat[fromFormat.length - 1]);
+  const result = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    object[fromFormat[i]] = splitDate[i];
+  }
+
+  if (object.hasOwnProperty('YYYY')) {
+    object.YY = object.YYYY.slice(2);
+  }
+
+  if (object.YY >= 30) {
+    object.YYYY = '19' + object.YY;
+  } else {
+    object.YYYY = '20' + object.YY;
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    result.push(object[toFormat[i]]);
+  }
+
+  return result.join(toFormat[toFormat.length - 1]);
```

Comment:
There is no check to ensure that 'YY' or 'YYYY' is included in the 'fromFormat' before accessing these properties on the 'object'. This could lead to accessing undefined properties and incorrect results. Ensure that you only access and manipulate the date parts that are specified in the 'fromFormat'.